### ISSUE TITLE
TypeError in HTTPSmsProvider

### ISSUE THREAD

**Issue by cornelinux** [privacyidea/privacyidea#889]
The HTTPSmsProvider will through a TypeError exception if a timeout is specified in the HTTP SMS Gateway.

This is due to these settings:
https://github.com/privacyidea/privacyidea/blob/master/privacyidea/lib/smsprovider/HttpSMSProvider.py#L91
https://github.com/privacyidea/privacyidea/blob/master/privacyidea/lib/smsprovider/HttpSMSProvider.py#L75

The timeout is saved as a string in the database and thus will be passed as string to ``requests.get`` or ``request.post``.

We need cast the timeout to an integer or float.

# Temporary workaround

You can delete the timeout in the HTTP Gateway config. A default value of ``3`` will be used.

**Comment by cornelinux** [comment]
The code seemed to be added on Nov 1st. 
https://github.com/privacyidea/privacyidea/commit/14140faf63c9ffad774c3f0a86be6e74d623abd4#diff-f534f3bf7b9ffc8acb661fd353622a86

So this is indeed only relevant for version 2.21 and 2.21.1

**Comment by fredreichbier** [comment]
I merged #890 (even though I couldn't test it as I don't have a HTTP SMS gateway -- but it looked good to me).
On a related note: [This line](https://github.com/privacyidea/privacyidea/blob/branch-2.21/privacyidea/lib/smsprovider/HttpSMSProvider.py#L87) looks to me like whatever we enter in ``CHECK_SSL`` gets passed to ``requests`` as a string. ``parameters`` defines a choice between ``"yes"`` and ``"no"`` for its value. Wouldn't that mean that we cannot ever disable the SSL check, as either the string ``"yes"`` or the string ``"no"`` are passed to requests?

**Comment by cornelinux** [comment]
CHECK_SSL = no means disable SSL check.

If the URL contains a HTTP shema, ssl is not checked anyways.

**Comment by fredreichbier** [comment]
Yes, I'm just not sure if the line I referenced above reads the values correctly. Shouldn't that be
```
 ssl_verify = self.config.get('CHECK_SSL',  'yes') == 'yes'
```
analogously to [here](https://github.com/privacyidea/privacyidea/blob/branch-2.21/privacyidea/lib/smsprovider/HttpSMSProvider.py#L71)?

**Comment by cornelinux** [comment]
"self.smsgateway" is the new configuration (current) that saves generic configs with "yes".

The "else" branch is an old, depricated branch, with a specif config, that is not touched anymore.

**Comment by cornelinux** [comment]
fixed in b7243070ff020218907f852ee09959a9115d4668
